Expensify/App
# Add finallyData as alternative to successData and failureData

### Details
Adds the option of using `finallyData` when `successData` and `failureData` are identical as discussed [here](https://expensify.slack.com/archives/C03TQ48KC/p1702350560750829). Updated `RequestNewValidateCode` to use this format for easy testing 

### Fixed Issues
$ https://github.com/Expensify/App/issues/33994
PROPOSAL:

### Tests
1. Try to log into an account on New Expensify using a magic code 
2. Confirm you can successfully log in 
3. Open up your JS console and navigate to Application > OnyxDB > IndexDB > keyvaluepairs 
4. Confirm you don't see an entry from you don't see `loadingForm` data for the `account` onyxkey


### Offline tests
N/A

### QA Steps

Same as test steps 


### PR Author Checklist

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>
